### DISCUSSION TITLE
How to insert template that keep indentation

### DISCUSSION THREAD

**Discussion by laofik219** [SilentVoid13/Templater#1514] [Help] [answered]
I want to insert this template in a list and have consistent indentation like on copy/paste

```markdown
#log/effort on [[<% tp.file.cursor() %>]]
	- I did
		- 
	- Next
		- 
``` 

and have this result

```markdown
- test 
    - test
        - #log/effort on [[/*cursor*/]]
	    - I did
	        - 
	    - Next
	        - 
```

Currently, the indent is like absolute.

I have tried: 
- different indent 
- outputting the string with `tR` but I wasn't able to place the cursor
- using prompt or suggester, but I prefer to write in the file for autocomplete and other plugin

**Comment by Zachatoo** [comment, marked as answer]
You may need to adjust the numbers slightly, but here's a proof of concept.

```js
<%*
const { ch } = tp.app.workspace.activeEditor.editor.getCursor();
-%>
#log/effort on [[<% tp.file.cursor() %>]]
<% " ".repeat(ch + 2) %>- I did
<% " ".repeat(ch + 4) %>- 
<% " ".repeat(ch + 2) %>- Next
<% " ".repeat(ch + 4) %>- 
```

**Reply by laofik219** [reply]
thx a lot
